pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -111,7 +111,9 @@ func (volumes *VolumesV2) createVolume(opts VolumeCreateOpts) (string, error) {
 func (volumes *VolumesV1) getVolume(diskName string) (Volume, error) {
 	var volume_v1 volumes_v1.Volume
 	var volume Volume
-	err := volumes_v1.List(volumes.blockstorage, nil).EachPage(func(page pagination.Page) (bool, error) {
+
+	opts := &volumes_v1.ListOpts{Name: diskName}
```

Comment:
The logic below looks for either name == diskName *or* the ID contains diskName.  That latter case sounds amazingly terrible, but are we confident we don't need it? (and if so, can we remove it in this PR to make it clear that that logic isn't going to trigger anymore?)